File: src/test/java/org/example/eventhub/mapper/LocationMapperTest.java
package org.example.eventhub.mapper;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.example.eventhub.dto.location.LocationCreateRequest;
import org.example.eventhub.dto.location.LocationResponseLong;
import org.example.eventhub.entity.Location;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class LocationMapperTest {

    private final LocationMapper mapper = new LocationMapper();

    private Location entity;
    private LocationCreateRequest createRequest;

    @BeforeEach
    void setUp() {
        entity = new Location("Moscow", "Tverskaya", "10", "near metro");
        createRequest = new LocationCreateRequest("Saint Petersburg", "Nevsky", "5", "main entrance");
    }

    @Test
    void toLongDto_shouldReturnCorrectDto() {
        LocationResponseLong expected =
                new LocationResponseLong("Moscow", "Tverskaya", "10", "near metro");

        assertEquals(expected, mapper.toLongDto(entity));
    }

    @Test
    void toEntity_shouldMapCreateRequestCorrectly() {
        Location result = mapper.toEntity(createRequest);

        assertEquals("Saint Petersburg", result.getCity());
        assertEquals("Nevsky", result.getStreet());
        assertEquals("5", result.getHouse());
        assertEquals("main entrance", result.getAdditionalInfo());
    }
}

File: src/test/kotlin/org/example/eventhub/mapper/LocationMapperTest.kt
package org.example.eventhub.mapper

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import org.example.eventhub.dto.location.LocationCreateRequest
import org.example.eventhub.dto.location.LocationUpdateRequest
import org.example.eventhub.entity.Location

class LocationMapperTest : StringSpec({

    val mapper = LocationMapper()

    "toEntity маппит LocationCreateRequest в Location" {
        val dto = LocationCreateRequest(
            city = "Moscow",
            street = "Tverskaya",
            house = "1",
            additionalInfo = "Near metro"
        )

        val entity = mapper.toEntity(dto)

        entity.city shouldBe "Moscow"
        entity.street shouldBe "Tverskaya"
        entity.house shouldBe "1"
        entity.additionalInfo shouldBe "Near metro"
    }

    "toEntity маппит LocationUpdateRequest в Location" {
        val dto = LocationUpdateRequest(
            city = "SPB",
            street = "Nevsky",
            house = "10",
            additionalInfo = null
        )

        val entity = mapper.toEntity(dto)

        entity.city shouldBe "SPB"
        entity.street shouldBe "Nevsky"
        entity.house shouldBe "10"
        entity.additionalInfo shouldBe null
    }

    "toLongDto маппит entity в dto" {
        val entity = Location(
            city = "Moscow",
            street = "Tverskaya",
            house = "1",
            additionalInfo = "Near metro"
        )

        val dto = mapper.toLongDto(entity)

        dto!!.city shouldBe "Moscow"
        dto.street shouldBe "Tverskaya"
        dto.house shouldBe "1"
        dto.additionalInfo shouldBe "Near metro"
    }

    "toLongDto возвращает null если entity null" {
        val dto = mapper.toLongDto(null)
        dto shouldBe null
    }
})
